public/js/util/defer.js

```diff
@@ -0,0 +1,16 @@
+"use strict";
+
+function defer() {
+  let resolve, reject;
+  const promise = new Promise(function() {
+    resolve = arguments[0];
+    reject = arguments[1];
+  });
+  return {
+    resolve: resolve,
+    reject: reject,
+    promise: promise
+  };
+}
+
+module.exports = defer;
```

Comment:
Yeah I don't have a good feeling about it. I think the most important thing is that we write code that is similar to other devtools code, and it doesn't matter where these modules live. We may want to merge `util` and `lib` somehow at some point.